base/regex.jl

```diff
@@ -459,7 +459,18 @@ match(r::Regex, s::AbstractString, i::Integer) = throw(ArgumentError(
 ))
 
 findnext(re::Regex, str::Union{String,SubString}, idx::Integer) = _findnext_re(re, str, idx, C_NULL)
-
+findprev(re::Regex, str::Union{String,SubString}, idx::Integer) = begin
+    if idx > nextind(str,lastindex(str)) || idx < prevind(str,firstindex(str))
+        throw(BoundsError())
+    end
+    cur_ind = idx
+    ans = nothing
+    while ans === nothing && cur_ind >= firstindex(str)
+        ans = findnext(re,str,cur_ind)
+        cur_ind -= 1
+    end
```

Comment:
I don't think this will find correct matches of repeated entities (such as `r"\w+"`. For that, I think you need to always call `findfirst`, but with the option `PCRE2_ENDANCHORED` set, and to then iteratively decrease the length of the string. That approach may fail for patterns with lookaheads however (such as `$`, `\B`, etc.).  There is also some ambiguity about whether it is permitted for results with `overlap=true` (c.f. `count`) for findprev which is not an issue for findnext.

(minor nit this would need `cur_ind = prevind(str, 1)`, not `cur_ind -= 1`)